addons/pos_self_order/models/product_product.py

```diff
@@ -32,12 +36,18 @@ def _filter_applicable_attributes(self, attributes_by_ptal_id: Dict) -> List[Dic
             if attributes_by_ptal_id.get(id) is not None
         ]
 
-    def _get_attributes(self, pos_config_sudo: PosConfig) -> List[Dict]:
+    def _get_attributes(
+        self, pos_config_sudo: PosConfig, attributes=None
+    ) -> List[Dict]:
         self.ensure_one()
 
-        attributes = self._filter_applicable_attributes(
-            self.env["pos.session"]._get_attributes_by_ptal_id()
-        )
+        if attributes is None:
+            attributes = self.env["pos.session"]._get_attributes_by_ptal_id()
+        else:
+            # Performance trick to avoid unnecessary calls to _get_attributes_by_ptal_id()
+            # Needs to be deep-copied because attributes is potentially mutated
+            attributes = deepcopy(attributes)
```

Comment:
don't deep copy here the whole "attributes" dictionary. it'll be more efficient to deep copy the result of `_filter_applicable_attributes`.